types/mjml-core/index.d.ts

```diff
@@ -133,11 +130,23 @@ export interface MJMLParseError {
     formattedMessage: string;
 }
 
-export interface MJMLJsonObject {
+export type MJMLJsonObject = MJMLJsonNonEndingTag | MJMLJsonEndingTag | MJMLJsonEndingTagWithoutContent;
+
+export interface MJMLJsonNonEndingTag {
+    tagName: string;
+    attributes: object;
+    children: MJMLJsonObject[];
+}
+
+export interface MJMLJsonEndingTag {
+    tagName: string;
+    attributes: object;
+    content: string;
+}
+
+export interface MJMLJsonEndingTagWithoutContent {
```

Comment:
Just so I'm clear, these would be examples of each JSON interface respectively,
```html
<mj-section>
  <mj-column>
    ...
  </mj-column>
</mj-section>
```

```html
<mj-text>
  My text content here
</mj-text>
```

```html
<mj-image />
```

Is this correct?

Reply:
Correct! Do you have a suggestion for better names to make this more clear? Or is there a comment you think would be valuable?

Reply:
What do you think about the following?

`MJMLJsonWithChildren`
`MJMLJsonWithContent`
`MJMLJsonSelfClosingTag`

Reply:
Thanks, Great suggestion! Done

Reply:
👍